### PR TITLE
feat: add hdf5 support

### PR DESCRIPTION
**Pull request by egorchakov**
This PR adds `hdf5` frame/table readers and an example for [Table Setup from Dishwasher](https://sites.google.com/view/il-for-mm/datasets#h.cq0r3rd5nr9m). 

image

NOTE: Unfortunately, `polars` doesn't natively support `hdf5` (https://github.com/pola-rs/polars/issues/3520), and `pandas`/`vaex` didn't work either for the example dataset -- hence the raw iteration.

### CODE REVIEW

File: examples/config_templates/visualize.yaml [hunk 1]
```diff
@@ -14,10 +14,10 @@ dataloader:
     _target_: rbyte.utils.dataloader.collate_identity
     _partial_: true
 
-  num_workers: 1
-  pin_memory: false
-  persistent_workers: true
-  multiprocessing_context: forkserver
+  # num_workers: 1
+  # pin_memory: false
+  # persistent_workers: true
```
Comment (sandhawalia) [inline examples/config_templates/visualize.yaml:19]: remove ? or commented by error

File: src/rbyte/io/table/hdf5/reader.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        config = self._config.model_dump_json()
+        # roundtripping json to work around https://github.com/pydantic/pydantic/issues/7424
+        config_str = json.dumps(json.loads(config), sort_keys=True)
+
+        return digest(config_str)
+
+    @override
+    def read(self, path: PathLike[str]) -> Mapping[str, pl.DataFrame]:
+        dfs: Mapping[str, pl.DataFrame] = {}
+
```
Comment (sandhawalia) [inline src/rbyte/io/table/hdf5/reader.py:53]: no support for hdf5 in polars leads to this nested nest of nests 😮‍💨

**Review by sandhawalia — APPROVED** [review]

**Comment by sandhawalia** [comment]
LGTM. As we are now upto 4 dataset formats. Shall we add some supporting docs f.e HOWTO.md

File: examples/config_templates/visualize.yaml (hunk 1 above)
Comment (egorchakov) [reply examples/config_templates/visualize.yaml:19]: accidental, good catch

**Comment by egorchakov** [comment]
> Shall we add some supporting docs f.e HOWTO.md

lets tee it up for the next release

**Pull request merged**